scene/main/scene_tree.h

```diff
@@ -133,9 +133,18 @@ class SceneTree : public MainLoop {
 		bool operator<(const UGCall &p_with) const { return group == p_with.group ? call < p_with.call : group < p_with.group; }
 	};
 
-	// Safety for when a node is deleted while a group is being called.
-	int call_lock = 0;
-	HashSet<ObjectID> call_skip; // Skip erased nodes. Store ID instead of pointer to avoid false positives when node is freed and a new node is allocated at the pointed address.
+	// Safety for when a node is deleted during a group call.
+	// For each nested group call the removed nodes are tracked separately (in a stack) because in different group calls the same
+	// pointer can be used for skipping different nodes (in case a node is freed and a new node is allocated at the same address).
+	// Checks for skipping are meant to be done by pointer (hence it's HashMap's key) as that's what groups store (cheaper than IDs?).
+	// To ensure such by pointer checks are correct, IDs of the removed nodes are stored as well (as values) so they can be used to
+	// update `call_skip` properly when a node is being added/removed to/from the tree.
+	Vector<HashMap<Node *, ObjectID>> call_skips_stack;
+	int call_skips_stack_size = 0; // Instead of shrinking the stack, we just `clear()` the popped HashMaps and keep track of the actual stack size.
```

Comment:
Not sure about this not shrinking part though. When testing it in the editor the stack was getting only to the size of 2 (which means there was only one nested group call made from some another group call). Maybe it could get bigger even in the editor but I wouldn't expect it to get way bigger. Also I'd say it getting big because of the user is also unlikely (unless they'd do some stack-overflowing type of recursion). So it's probably  fine to just keep the stack at the maximum size it got to, instead of continously resizing it.